src/main/java/com/yelp/nrtsearch/server/luceneserver/field/ContextSuggestFieldDef.java

```diff
@@ -64,22 +60,26 @@ public String getType() {
   @Override
   public void parseDocumentField(
       Document document, List<String> fieldValues, List<List<String>> facetHierarchyPaths) {
-    if (fieldValues.size() == 1) {
-      ContextSuggestFieldData csfData =
-          GSON.fromJson(fieldValues.get(0), ContextSuggestFieldData.class);
-      CharSequence[] contexts =
-          csfData.getContexts().toArray(new CharSequence[csfData.getContexts().size()]);
-      ContextSuggestField csf =
-          new ContextSuggestField(getName(), csfData.getValue(), csfData.getWeight(), contexts);
-      document.add(csf);
-      if (isStored()) {
-        document.add(new FieldWithData(getName(), fieldType, fieldValues.get(0)));
-      }
-    } else {
-      throw new IllegalArgumentException("Context Suggest Field can only index exactly one value");
+    for (String fieldValue : fieldValues) {
+      parseFieldValueToDocumentField(document, fieldValue);
     }
```

Comment:
Probably not a big deal right now, but it would be good to fail if multiple values are provided when the field does not have multi value set true.

Reply:
resolved, added exception similar to other field def classes